File: GetHTML.php
<?php
// this pgm uses another python pgm to fetch a web page and render in pure html, then displays
function geturl($url)
{
$base=$_SERVER['DOCUMENT_ROOT'];
$cmd='sudo  '.$base.'cgi-bin/GetPage.sh '.$url;
//DEBUG error_log("cmd=".$cmd,3,"/tmp/err.txt");
ob_start();
passthru($cmd);
$html = ob_get_clean(); 
return $html;
}
if(__FILE__ ==$_SERVER['SCRIPT_FILENAME'])   // if this script called directly from server,
    {print(geturl($_GET['url']));}					//  display page from url passed in this request 
?>

File: GetHTML.py
#!/usr/bin/python
## WIP - not tested!
## get web page using GetPage.py;  this is a wrapper intended for apache web server use
# Turn on debug mode.
from GetPage import *
import cgitb
cgitb.enable()

# Print necessary headers.
print("Content-Type: text/html")
print()
url="https://www.xe.com/currency/nzd-new-zealand-dollar"
from pyvirtualdisplay import Display
# display = Display(visible=0, size=(800, 600))
with Display(visible=0, size=(800, 600)) as display:  ## with breaks down display on exit
          view = Render()
          view.batch=True
          view.load(QUrl(url)) ## get url from passed args
          view.app.exec_()
          display.sendstop()
          print(view.html)
quit()
